app/models/post.rb

```diff
@@ -0,0 +1,32 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'

```

Comment:
[OPTIONAL] Rails adds foreign keys as default. So no need for adding foreign keys for other associations except the associations to the author_id column. So kindly consider removing the foreign_keys here.